Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -92,6 +92,17 @@ public override object DeserializeObject(object value, Type type)
                 {
                     if (ReflectionUtils.GetTypeInfo(type).IsEnum)
                     {
+                        //first try to get all custom attributes
```

Comment:
yeah im not saying there is a better way of doing it or that it does/doesn't have performance impacts, just asking the question and looking for more opinions.  

Im not a an of hardcoded strings or a new attribute, if anything had to be done, I would probably do similar to how the serializer pre-caches all the constructors of the model objects, and do some sort of pre-cache of enum items and their value (custom or default), then it would be a quick look up of that list here in the deserializer.  

It may not even be necessary though, which is why i refer to @shiftkey